applications/Cook Protocol.md

```diff
@@ -0,0 +1,151 @@
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. 
+| 1. | Frontend | We will implement frontend using React.js so that:  1. Users can connect to the polkadot wallet.  2. The fund manager can initialize a fund, and allocate assets via Polkaswap.  3. The investor can browse the fund list, check the performance details of a fund, and invest in a fund. |  
+| 2. | Substrate | Create a placeholder parachain which frontend can interact with. |    
+| 3. | Docker | We will provide a dockerfile to demonstrate the full functionality |
+
+### Milestone 2 Fund Investment & Management Implementation
+* **Estimated Duration:** 3 weeks
+* **FTE:**  2
+* **Costs:** 6k DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Substrate nodes. Once the node is up, it will be possible to send test transactions that will show how the new functionality works. |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. 
+| 1. | Fund management | We will implement the API for fund managers to create a fund, manage a fund and claim management fees. |  
+| 2. | Fund investment | We will implement the API for investors to invest in a fund to mint ckToken, and burn ckToken to withdraw their investments. |    
+| 3. | Docker | We will provide a dockerfile to demonstrate the full functionality |
+
+### Milestone 3 Integration with Polkaswap
+* **Estimated Duration:** 2 weeks
+* **FTE:**  2
+* **Costs:** 4k DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Substrate nodes. Once the node is up, it will be possible to send test transactions that will show how the new functionality works. |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. 
+| 1. | Cross-chain asset exchange | We will exchange assets on Polkaswap using HRMP. This will enable fund managers to allocate assets for the fund. |    
```

Comment:
What are you technically implementing here? Also is polkaswap actually directly implemented as a parachain or is it only a bridge to a different chain, which will run polkaswap? In general it’s difficult at this stage to know how polkaswap will be implemented. Potentially they don’t even get the parachain slot, etc. So ideally this is part of the delivery process, but the deliverables are based on concrete technical implementations.

Reply:
We realized that Polkaswap is far from ready. In order to mitigate uncertainty, we decided to use Centrifuge (https://centrifuge.io) instead, which is based on a Substrate Node to act as a parachain bridge. This project is an open-source project and its code can be found here: https://github.com/centrifuge/centrifugechain

Reply:
Makes sense